Tests/Functional/Listener/FormRequestTest.php

```diff
@@ -0,0 +1,238 @@
+
+class FormRequestTest extends SuluTestCase
+{
+
+    /**
+     * @var \Symfony\Bundle\FrameworkBundle\KernelBrowser
+     */
+    private $client;
+
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function testPageRendersWithoutForm()
+    {
+        $this->inititalize();
+
+        $this->updateHomePage(null);
+
+        $this->client->request('GET', '/');
+//        $this->assertResponseIsSuccessful();
+        $response = $this->client->getResponse();
+        $this->assertHttpStatusCode(200, $response);
+    }
+
+    public function testPageRendersWithSetForm()
+    {
+        $this->inititalize();
+
+        $form = $this->createFullForm();
+        $this->updateHomePage($form);
+        $this->doSendForm($form);
+
+        $mailCollector = $this->client->getProfile()->getCollector('swiftmailer');
+        // 2 messages should be send 1 to admin and 1 to email
+        $this->assertSame(2, $mailCollector->getMessageCount());
+
+        $this->assertEmailCount(0);
+    }
+
+    public function testDoesNotMailWhenNullHandlerSet()
+    {
+        $_ENV['MAIL_HANDLER'] = 'NullHelper';
+        $this->inititalize();
+
+        $form = $this->createFullForm();
+        $this->updateHomePage($form);
+        $this->doSendForm($form);
+
+        $mailCollector = $this->client->getProfile()->getCollector('swiftmailer');
+        // 2 messages should be send 1 to admin and 1 to email
+        $this->assertSame(0, $mailCollector->getMessageCount());
+
+        $this->assertEmailCount(0);
+    }
+
+    public function testUsesMailerWhenMailerHandlerSet()
+    {
+        $_ENV['MAIL_HANDLER'] = 'MailerHelper';
+        $this->inititalize();
+
+        $form = $this->createFullForm();
+        $this->updateHomePage($form);
+        $this->doSendForm($form);
+
+        $mailCollector = $this->client->getProfile()->getCollector('swiftmailer');
+        $this->assertSame(0, $mailCollector->getMessageCount());
+
+        $this->assertEmailCount(2);
+    }
+    /**
+     * @param Form $form
+     * @throws \Sulu\Component\DocumentManager\Exception\DocumentManagerException
+     */
+    private function updateHomePage(Form $form = null): void
+    {
+        /* @var $suluDocumentManager \Sulu\Component\DocumentManager\DocumentManagerInterface */
+        $suluDocumentManager = static::getContainer()->get('sulu_document_manager.document_manager');
+
+        /* @var $homePage \Sulu\Bundle\PageBundle\Document\HomeDocument */
+        $homePage = $suluDocumentManager->find('/cmf/sulu-io/contents');
+        $homePage->setResourceSegment('/');
+        $homePage->getStructure()->bind([
+            'form' => $form ? $form->getId() : null,
+            'url' => '/'
+        ]);
+
+        $suluDocumentManager->publish($homePage, 'de');
+        $suluDocumentManager->persist($homePage, 'de');
+        $suluDocumentManager->flush();
+    }
+
+
+    private function createFullForm(): Form
```

Comment:
Would move this into a `CreateFormTrait` in Tests/Functional/Trait namespace.

Reply:
I turned this into a fixture instead since I think that is more close to what it should be. Of course happy to still make it into a trait.

I remember there was a nice way to apply fixtures but well i came up with this :)